### PR TITLE
Add example crate

### PR DESCRIPTION
**Pull request by SUPERCILEX**
Closes https://github.com/assert-rs/trycmd/issues/146

I'm not quite sure how or if you want to do CI integration since the crate isn't part of the workspace.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,5 +1,6 @@
 [workspace]
 members = ["crates/*"]
+exclude = ["examples/demo_trycmd"]
```
Comment (epage) [inline Cargo.toml:3]: I believe you can avoid the exclude if you put an empty `[workspace]` inside the crate

File: examples/demo_trycmd/README.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+Must supply exactly one argument.
+
+$ GOODBYE=true simple World
+Goodbye World!
+
+```
+
+Sometimes, your test might include output that is non-deterministic. When that's the case, you can
+use variables to replace those values. In our `tests/trycmd.rs`, we've defined a
+variable `[REPLACEMENT]` such that whenever the value `non-deterministic-value` appears, it will be
```
Comment (epage) [inline examples/demo_trycmd/README.md:39]: I think referring this to a runtime-generated value might be more accurate as truly non-deterministic values can't be determined by the test function to pass in

File: src/lib.rs [hunk 3]
```diff
@@ -107,23 +107,7 @@
...
-//! ```
-//!
-//! You can verify a code block like:
-//! ~~~md
-//! ```console
-//! $ my-cmd
-//! Hello world
-//!
-//! ```
-//! ~~~
```
Comment (epage) [inline src/lib.rs:126]: I lean towards us having both.  The inline example first and then a "for a more complete example, see ..."

Comment (SUPERCILEX) [reply src/lib.rs:126]: Sounds good.

File: examples/demo_trycmd/README.md (hunk 2 above)
Comment (SUPERCILEX) [reply examples/demo_trycmd/README.md]: Done.

File: src/lib.rs [hunk 4]
```diff
@@ -108,23 +108,23 @@
...
-//! You can verify a code block like:
+//! You get the following:
 //! ~~~md
 //! ```console
-//! $ my-cmd
-//! Hello world
+//! $ my-cmd --print 'Hello World'
+//! Hello
 //!
 //! ```
```
Comment (epage) [inline src/lib.rs:122]: Please revert these changes; they are unrelated to the current change and, while there might be some discussion on iteration on it, I'd rather we decouple it from this specific PR.

Comment (SUPERCILEX) [reply src/lib.rs]: Dang, sorry. This and the imports have been unintentional, thanks for catching it. Fixed my rebase.

**Comment by epage** [comment]
Thanks!
